docs/new-editor.md

```diff
@@ -0,0 +1,163 @@
+automatically to Bintray by our Travis CI. There are no stable or officially
+supported releases of metals.
+
+The following Java options are recommended:
+
+* `-XX:+UseG1GC -XX:+UseStringDeduplication`: to reduce memory consumption from
+  navigation indexes. May not be necessary in the future.
+
+Refer to the coursier documentation for how to build a fat jar or configure java
+parameters.
+
+## Working directory
+
+The server needs to be started in the same directory matching the `rootUri`
+parameter of the `initialize` request. Goto definition and other
+SemanticDB-features will not work if the working directory does not match the
+root directory of the build.
+
+NOTE. Discarding `rootUri` is not compliant with LSP, consider contributing to
+[#216][] if you are affected by this issue.
+
+## File watching
+
+Metals delegates file watching to the editor client by listening for
+[`workspace/didChangeWatchedFiles`][] notifications. The server expects the
+client to send notifications for changes to files matching the following
+patterns
+
+* `.metals/buildinfo/**/*.properties`: build metadata to enable goto definition
+  for the classpath, completions with the presentation compiler and refactoring
+  with Scalafix.
+* `**/*.semanticdb`: artifacts produced by the semanticdb-scalac compiler plugin
+  during batch compilation in the build. See
+  [here](https://github.com/scalameta/scalameta/blob/master/semanticdb/semanticdb3/semanticdb3.md)
+  to learn more about SemanticDB. These files are required for goto definition,
+  find references, hover and Scalafix to work.
+
+See the VS Code plugin
+[clientOptions](https://github.com/scalameta/metals/blob/fb166f1d81eb77ebd9c6b3ee95e65fb58a907eec/vscode-extension/src/extension.ts#L44-L54)
+for an example of how file watching is handled in VS Code.
+
+The metals server scans the workspace directory for these file patterns on
+startup regardless of file watching notifications. This means goto definition
+and other functionality may work to begin with, but quickly degrade for longer
+running sessions.
+
+NOTE. The metals server does not use `DidChangeWatchedFilesRegistrationOptions`
+to register these particular patterns, consider contributing to [#255][] if you
+are affected by this issue.
+
+## Configuration
+
+Metals uses
+[`workspace/didChangeConfiguration`](https://microsoft.github.io/language-server-protocol/specification#workspace_didChangeWatchedFiles)
+notifications to allow end-users to control behavior of the server. The editor
+client is expected to send a `workspace/didChangeConfiguration` notification
+containing user configuration right after the `initialized` notification.
+
+A full list of supported configuration options can be found in the VS Code
+plugin
+[package.json](https://github.com/scalameta/metals/blob/master/vscode-extension/package.json)
+manifest. Some configuration options are required by the server while others
+should be handled by the client.
```

Comment:
I would link to https://github.com/scalameta/metals/blob/master/metals/src/main/scala/scala/meta/metals/Configuration.scala for the server configuration (it's fine to also link to the vscode implemenation as an example)

Also, they're not properly "required", since there are defaults for everything. The client can not send any configuration and the server will work nonetheless.

Reply:
Good idea. Updated.